File: features/pages/login_page.rb
class Pagelogin
  def informarlogin
    @driver.find_element(:accessibility_id, "email")
  end

  def informarSenha
    @driver.find_element(:accessibility_id, "senha")
  end

  def selecionarEntrar
    @driver.find_element(:xpath, "//android.view.ViewGroup[@content-desc=\"entrar\"]/android.view.ViewGroup")
  end

  def informarCodigo
    @driver.find_element(:accessibility_id, "codigo")
  end
end

File: UITest-src/main/java/Pages/LogInPage.java
package Pages;

import Infrastructure.BasePage;
import io.appium.java_client.MobileBy;
import io.appium.java_client.MobileElement;

public class LogInPage extends BasePage {

    // Element Mapper
    //================================================================================

    public MobileElement EmailInput() {
        return FindElement(MobileBy.xpath("//android.widget.EditText[@text='E-mail']"));
    }

    public MobileElement SenhaInput() {
        return FindElement(MobileBy.xpath("//android.widget.EditText[@text='Senha']"));
    }

    public MobileElement EntrarButton() {
        return FindElement(MobileBy.xpath("//android.widget.TextView[@text='Entrar']"));
    }

    public MobileElement ErrorMessage() {
        return FindElement(MobileBy.xpath("//android.widget.TextView[@content-desc='lognFail']"));
    }

    //================================================================================

    // Element Iterators
    //================================================================================

    public void InsertEmail(String email) {
        EmailInput().sendKeys((email));
    }

    public void InsertSenha(String senha) {
        SenhaInput().sendKeys((senha));
    }

    public void ClickEntrar() {
        EntrarButton().click();
    }

    //================================================================================

    // Page Behaviors
    //================================================================================

    public void Login(String email, String senha) {
        InsertEmail(email);
        InsertSenha(senha);
        ClickEntrar();
    }
}
